frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,220 @@
+    const {filterValue} = this.state;
+    if (namespace !== prevProps.namespace || clusterserviceclasses !== prevProps.clusterserviceclasses) {
+      this.createCatalogData();
+    } else if (filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems
+      });
+    }
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      return fuzzy(filterString, _.get(item, 'metadata.name', ''));
+    });
+  }
+
+  createCatalogData() {
+    const {clusterserviceclasses, imagestreams, loaded} = this.props;
+
+    if (!loaded) {
+      return;
+    }
+
+    const clusterServiceClassItems = this.normalizeClusterServiceClasses(clusterserviceclasses.data, 'ClusterServiceClass');
+    const imagestreamsItems = this.normalizeImagestreams(imagestreams.data, 'ImageStream');
+
+    const items = [
+      ...clusterServiceClassItems,
+      ...imagestreamsItems
+    ];
+
+    const filteredItems = this.filterItems(items);
+    this.setState({items, filteredItems});
+  }
+
+  normalizeClusterServiceClasses(serviceClasses, kind) {
+    const filteredServiceClasses = _.filter(serviceClasses, serviceClass => {
+      return !serviceClass.status.removedFromBrokerCatalog;
+    });
+
+    return _.map(filteredServiceClasses, serviceClass => {
+      const tileName = serviceClassDisplayName(serviceClass);
+      const iconClass = getCSCIcon(serviceClass);
+      const imgUrl = getCSCImage(serviceClass, iconClass);
+      const tileIconClass = imgUrl ? null : iconClass;
+      const tileImgUrl = imgUrl ? imgUrl : null;
+      const tileDescription = _.get(serviceClass, 'spec.description');
+      const tileProvider = _.get(serviceClass, 'spec.externalMetadata.providerDisplayName');
+
+      return {
+        resource: serviceClass,
+        kind,
+        tileName,
+        tileIconClass,
+        tileImgUrl,
+        tileDescription,
+        tileProvider
+      };
+    });
+  }
+
+  normalizeImagestreams(imagestreams, kind) {
+    imagestreams = _.map(imagestreams, imagestream => {
+      // A valid specTag has a 'builder' tag. no 'hidden' tag, and exists in 'status.tags'.
+      const builderSpecTagName = getBuilderSpecTagName(imagestream);
+      if (builderSpecTagName) {
+        const tileName = _.get(imagestream, ['metadata', 'annotations', 'displayName']) || imagestream.metadata.name;
+        // TODO: icon, description and provider should be based on selected version?
+        const tileIconClass = getImageStreamTagAnnotation(imagestream, 'iconClass', builderSpecTagName);
+        const tileDescription = getImageStreamTagAnnotation(imagestream, 'description', builderSpecTagName);
+        const tileProvider = getImageStreamTagAnnotation(imagestream, 'openshift.io/provider-display-name', builderSpecTagName);
```

Comment:
Why not have `getBuildSpecTagName` just return the tag instead of the name, then you can get annotations easily without the helper.

Reply:
Ok, I'll readdress after #534 is merged because I'll probably be using the methods there.